### DISCUSSION TITLE
How I Can Retrieve LightHouse Data Quality ?

### DISCUSSION THREAD

**Discussion by imamim** [bitcraze/discussions#663] [Q&A] [answered]
Hello, we aim to use the LightHouse deck to fly our drones and want to check if the LightHouse data is being properly received to determine the status of the drone (i.e., whether it is in any fault state or not,may be quality check done). We plan to achieve this using cflib-python. How can we do this?

**Comment by krichardsson** [comment, marked as answer]
You can perhaps use the logs that are displayed in the lighthouse tab in the client
[lighthouse.bsAvailable](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#lighthousebsavailable)
[lighthouse.bsReceive](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#lighthousebsreceive)
[lighthouse.bsActive](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#lighthousebsactive)
[lighthouse.bsCalUd](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#lighthousebscalud)
[lighthouse.bsCalCon](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#lighthousebscalcon)

**Comment by imamim** [comment]
Thanks! `lighthouse.status` variable solve our problem using SyncLogger. When the LightHouse data status is okay returns `2` and we test it.We verified it works.

But we have a some problems about `SyncCrazyflie.is_link_open()` function for again same purpose.The problem is that:

This function continues to return the value `True` even though the drone has lost power. We are not sure if we understand this function correctly. How can we detect if the communication is still up in real time?
